Source/WebCore/dom/Document.cpp

```diff
@@ -383,6 +384,33 @@ struct Document::PendingScrollEventTargetList {
 
 static const Seconds intersectionObserversInitialUpdateDelay { 2000_ms };
 
+class CallbackForContainIntrinsicSize : public NativeResizeObserverCallback {
+public:
+    void onResize(const Vector<Ref<ResizeObserverEntry>>&) final;
+};
+
+void CallbackForContainIntrinsicSize::onResize(const Vector<Ref<ResizeObserverEntry>>& entries)
+{
+    for (auto& entry : entries) {
+        if (auto* target = entry->target()) {
+            if (!target->isConnected()) {
+                target->document().unobserveForContainIntrinsicSize(*target);
+                target->clearLastRememberedSize();
+                continue;
+            }
+
+            auto* box = target->renderBox();
+            if (!box) {
+                target->document().unobserveForContainIntrinsicSize(*target);
+                continue;
+            }
+
+            auto contentBoxSize = entry->contentBoxSize().at(0);
+            target->setLastRememberedSize(ResizeObserverSize::create(contentBoxSize->inlineSize(), contentBoxSize->blockSize()));
```

Comment:
The test for `display: inline` not clearing the last remembered size seems to pass, but where are you handling that? Seems like this line would set it to 0x0.